bin/brew

```diff
@@ -9,20 +9,43 @@ BREW_FILE_DIRECTORY="$(quiet_cd "${0%/*}/" && pwd -P)"
 HOMEBREW_BREW_FILE="${BREW_FILE_DIRECTORY%/}/${0##*/}"
 
 HOMEBREW_PREFIX="${HOMEBREW_BREW_FILE%/*/*}"
-[[ -n "$HOMEBREW_PREFIX" && "$HOMEBREW_PREFIX" != "$HOMEBREW_BREW_FILE" ]] \
-  || HOMEBREW_PREFIX="/"
+if [[ -z "$HOMEBREW_PREFIX" || "$HOMEBREW_PREFIX" == "$HOMEBREW_BREW_FILE" ]]
+then
+  HOMEBREW_PREFIX="/"
+fi
 
 HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX"
 
 if [[ -L "$HOMEBREW_BREW_FILE" ]]
 then
+  HOMEBREW_BREW_FILE_LINK="1"
+fi
+
+if [[ -L "/usr/local/bin/brew" ]]
+then
+  HOMEBREW_USR_LOCAL_HOMEBREW_TEST="1"
+fi
+
+if [[ -n "$HOMEBREW_BREW_FILE_LINK" ||
+      -n "$HOMEBREW_USR_LOCAL_HOMEBREW_TEST" ]]
+then
   BREW_SYMLINK="$(readlink "$HOMEBREW_BREW_FILE")"
```

Comment:
> As it is now, the code doesn't do what it should be doing (if I'm reading it correctly).

To be more specific, if I have symbolic links

- `/usr/local/bin/brew -> /usr/local/homebrew/bin/brew` and
- `/opt/brewery/tests/bin/brew -> /opt/brewery/tests/repo/bin/brew`,

then invoking `/opt/brewery/tests/repo/bin/brew` or its symlink `/opt/brewery/tests/bin/brew` will incorrectly determine the Homebrew prefix to be `/usr/local`.